geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloBackedRasterOperations.scala

```diff
@@ -94,13 +102,36 @@ class AccumuloBackedRasterOperations(val connector: Connector,
     writeMutations(createMutation(raster))
   }
 
+  def getMosaicedRaster(query: RasterQuery, params: GeoMesaCoverageQueryParams) = {
+    val rasters = getRasters(query)
+    val (image, numRasters) = profile(RasterUtils.mosaicRasters(rasters,
+                                                        params.height.toInt,
+                                                        params.width.toInt,
+                                                        params.envelope,
+                                                        params.resX,
+                                                        params.resY), "mosaic")
+    val stat = RasterQueryStat(rasterTable,
+      System.currentTimeMillis(),
+      query.toString,
+      timings.time("planning"),
+      timings.time("scanning") - timings.time("planning"),
+      timings.time("mosaic"),
+      numRasters)
+    this.writeStat(stat, s"${rasterTable}_queries")
+    image
+  }
+
   def getRasters(rasterQuery: RasterQuery): Iterator[Raster] = {
-    //TODO: WCS: Abstract number of threads
-    val numQThreads = 20
-    val batchScanner = connector.createBatchScanner(rasterTable, authorizationsProvider.getAuthorizations, numQThreads)
-    val plan = queryPlanner.getQueryPlan(rasterQuery)
-    configureBatchScanner(batchScanner, plan)
-    adaptIterator(batchScanner.iterator)
+    profile({
+      val batchScanner = connector.createBatchScanner(rasterTable, authorizationsProvider.getAuthorizations, numQThreads)
+      val plan = profile(queryPlanner.getQueryPlan(rasterQuery), "planning")
+      configureBatchScanner(batchScanner, plan)
+      adaptIterator(batchScanner.iterator)}, "scanning")
```

Comment:
See above...  This would turn into 
```
 profile("scanning") {
  val batchScanner = connector.createBatchScanner(rasterTable, authorizationsProvider.getAuthorizations, numQThreads)
  val plan = profile(queryPlanner.getQueryPlan(rasterQuery), "planning")
  configureBatchScanner(batchScanner, plan)
  adaptIterator(batchScanner.iterator)
}
```